src/client/pages/Checkout/CheckoutPayment/CheckoutPayment.tsx

```diff
@@ -224,6 +224,7 @@ export const CheckoutPayment = ({
       },
     } as QuoteInput,
     validationSchema: getCheckoutDetailsValidationSchema(locale, textKeys),
+    validateOnMount: true,
```

Comment:
I'm not sure what the use case looks like, but isn't it generally a good idea to allow the user to get the chance to fill in information before showing the error? Now it will look like the user has entered incorrect information or tried to submit the form even before they've had a chance to enter anything? Or am I misinterpreting the flow?

I totally agree that we shouldn't submit when there are errors, but Formik validates on submit, so it won't send any requests if the form isn't valid.

Reply:
I agree with Simon that we should not show the error right away. That said, the validation on submit is not automatic I believe. We use the following piece of code in the `startSign` function in the old offer page to force validation. I think we should do something similar

```
const errors = await validateForm()

if (Object.keys(errors).length) {
  setSignUiState('FAILED')
  return
}
```

https://github.com/HedvigInsurance/web-onboarding/blob/62d429c13430dafe0a8616e5b6739bf715689e87/src/client/pages/Offer/Checkout/index.tsx#L392

Reply:
Perhaps we're not using Formik's submit then? It _should_ validate: https://formik.org/docs/guides/validation#when-does-validation-run

Reply:
☝️ yeah let's look into why that's not working out of the box

Reply:
@simonauner the "complete purchase" button didn't belong to that form so that's why it wasn't validating the inputs

Reply:
Ah, good find!